tests/foreman_api_test.py

```diff
@@ -0,0 +1,28 @@
+import urllib.parse
+
+import requests
+
+
+def _repo_url(repo, ssh_config):
+    return urllib.parse.urlunparse(urllib.parse.urlparse(repo['full_path'])._replace(netloc=ssh_config['hostname']))
```

Comment:
I recall having a discussion with @maximiliankolb about exposing the repo's full path URLs in Katello. I think we should file an issue to expose this in the API (and UI) for a specific content source. Then add a TODO here to use that API when it becomes available.

Any thoughts on that?

Reply:
What do you mean by "full path"? This attribute is the full path in the form of `https://foreman.example.com/pulp/content/blah/blah/blah`. It's just that I need to fuck around with the fact that `foreman.example.com` doesn't exist as a DNS name when the tests run and thus have this ugly replacement here.

Reply:
Ah. Should we just fix that instead then? I can take a stab at that, though probably better to capture it in an issue first and link it here as a todo

Reply:
I think we can "configure" it via https://github.com/theforeman/foreman-quadlet/blob/master/roles/foreman_proxy/templates/pulpcore.yaml.j2#L4

Reply:
pytest, today, runs on the GH runner directly (or on my laptop)

Reply:
Hmm ok. Perhaps leave a comment on why the hack is needed then?

Reply:
RE exposing the URLs to foreman-katello.example.com/pulp/content/...: Yes please. It's necessary if you use F+K as content source when registering non-RHEL hosts using global host registration because non-RHEL OS do not have (access to) `subscription-manager`. We are also (someday) working on a button in the WebUI to allow users to copy the Katello API endpoint for the associated GPG pub key.

See https://github.com/theforeman/foreman-documentation/pull/3380 and https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#Content-Views-in-Foreman_content-management